### PR TITLE
fix(auth, android): return credential for signin if phone auth has link collision

### PR DESCRIPTION
**Pull request by joaqo**
### Description

You can now upgrade anonymous users using phone auth on Android. Credit to @Shaninnik.

Fixes:
 - Fixes: #4911

### CODE REVIEW

File: packages/auth/android/src/main/java/io/invertase/firebase/auth/ReactNativeFirebaseAuthModule.java [hunk 1]
```diff
@@ -107,6 +108,10 @@ class ReactNativeFirebaseAuthModule extends ReactNativeFirebaseModule {
   private final HashMap<String, MultiFactorResolver> mCachedResolvers = new HashMap<>();
   private final HashMap<String, MultiFactorSession> mMultiFactorSessions = new HashMap<>();
 
+  // patching anonumous phone auth linkWithCredentials
```
Comment (mikehardy) [inline packages/auth/android/src/main/java/io/invertase/firebase/auth/ReactNativeFirebaseAuthModule.java:111]:
```suggestion
  // storage for anonymous phone auth credentials, used for linkWithCredentials
```

**Review by mikehardy — CHANGES_REQUESTED** [review]
Thanks for posting this - looks promising with some comments as indicated
I made them all in the form of suggestions, I'll apply them + reformat + push to the branch then do final review

I think this can go in though

**Review by mikehardy — DISMISSED** [review]
This looks reasonable, if it passes CI I can merge this in

**Comment by mikehardy** [comment]
Ah, I do love a thorough test suite - it picked something up on this one so it will need some further examination before merge:

```
      linkWithCredential()
        ✔ should link anonymous account <-> email account
23:32:06.039 detox[5958] ERROR: [WS_ERROR] The app has crashed, see the details below:

@Thread pool-106-thread-1(633):
java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Object.hashCode()' on a null object reference
	at io.invertase.firebase.auth.ReactNativeFirebaseAuthModule.promiseRejectLinkAuthException(ReactNativeFirebaseAuthModule.java:2244)
	at io.invertase.firebase.auth.ReactNativeFirebaseAuthModule.lambda$linkWithCredential$34(ReactNativeFirebaseAuthModule.java:1626)
	at io.invertase.firebase.auth.ReactNativeFirebaseAuthModule.$r8$lambda$Lodml2tmhgEN3Twg2KfjZTSQTN4(Unknown Source:0)
	at io.invertase.firebase.auth.ReactNativeFirebaseAuthModule$$ExternalSyntheticLambda1.onComplete(Unknown Source:4)
	at com.google.android.gms.tasks.zzi.run(com.google.android.gms:play-services-tasks@@18.1.0:1)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
	at java.lang.Thread.run(Thread.java:1012)
```

**Comment by joaqo** [comment]
This is fantastic, thank you so much @mikehardy for the fixes and the merge!

**Comment by joaqo** [comment]
Is there anywhere I can look to keep track for your release schedule so I know when to upgrade my dependencies? Thanks!

**Comment by mikehardy** [comment]
Unfortunately our release process doesn't formally make releases in GitHub so you can't really subscribe there, apologies. I usually release things about as soon as I merge them, this one just was held up a few days on CI issues, it's out now though

**Comment by joaqo** [comment]
No problem, yesterday I found you had a CHANGELOG file and noticed the new releases, so I upgraded our dependencies. Everythings working smoothly ✌️, thanks!

**Pull request merged**